### PR TITLE
Fix some panics found by the fuzzer

### PR DESCRIPTION
**Pull request by marcusklaas**
This addresses two, and adds a (disabled) test for a third one, which I hope to fix later.

### CODE REVIEW

**Comment by 5225225** [comment]
Is

```rust
#[test]
fn unwrap_none() {
    parse("*\t[][\n\t<p]>\n\t[]");
}
```

one of the panics addressed here (`thread 'unwrap_none' panicked at 'called `Option::unwrap()` on a `None` value', src/parse.rs:1083:46`)?

**Comment by marcusklaas** [comment]
It appears it is not! That example also fails without any options set. Nice find!

File: src/firstpass.rs [hunk 1]
```diff
@@ -457,276 +456,273 @@ impl<'a, 'b> FirstPass<'a, 'b> {
         let mut last_pipe_ix = start;
         let mut begin_text = start;
 
-        let (final_ix, brk) =
```
Comment (marcusklaas) [inline src/firstpass.rs:460]: This is unfortunately just `rustfmt` deciding that this entire expression needs to rewritten. Sorry for the noise! Although it does reduce the entire indentation level by one, which is nice. I'm just hoping it doesn't jump back on further modification...

**Comment by marcusklaas** [comment]
Okay, this now fixes the last of the known panics. I did another quick 15 min fuzz which did not find anything.

The low hanging fruit is officially gone!

**Comment by 5225225** [comment]
Cool! ~~When~~ *If* I find any more issues, I'll add them as new issues.

**Review by raphlinus — APPROVED** [review]
I did a quick skim and don't see anything that concerns me. I think increasingly we want to rely on tests and fuzzing to make sure it's doing the right thing, it's hard to reason about the logic. If there's anything that you feel needs special scrutiny, lemme know and I can take a closer look.

**Comment by marcusklaas** [comment]
Thanks for the review Raph!

I am so happy with the outcome of the fuzzer spree so far. It's not only some missing checks that were triggered, but real bugs that were just hard to find. The code has become better as a result :)

**Pull request merged**